core/blockchain.go

```diff
@@ -1480,13 +1462,18 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, er
 		// Falls through to the block import
 	}
 	switch {
+	// We're post-merge and the parent is pruned, try to recover the parent state
+	case !setHead && errors.Is(err, consensus.ErrPrunedAncestor):
+		log.Debug("Pruned ancestor", "number", block.Number(), "hash", block.Hash())
+		return it.index, bc.recoverAncestors(block)
+
```

Comment:
I think the we should try to keep the switch clauses a bit simpler, like this (and remove the clause right after)

```suggestion
	case errors.Is(err, consensus.ErrPrunedAncestor):
		if !setHead {
			// We're post-merge and the parent is pruned, try to recover the parent state
			log.Debug("Pruned ancestor", "number", block.Number(), "hash", block.Hash())
			return it.index, bc.recoverAncestors(block)
		}
		// First block is pruned, insert as sidechain and reorg only if TD grows enough
		log.Debug("Pruned ancestor, inserting as sidechain", "number", block.Number(), "hash", block.Hash())
		return bc.insertSideChain(block, it)
```